lib/ex_gram/bot/supervisor.ex

```diff
@@ -50,56 +50,39 @@ defmodule ExGram.Bot.Supervisor do
           other
       end
 
-    maybe_setup_commands(opts[:setup_commands], commands, token)
-
-    bot_info = maybe_fetch_bot(opts[:username], token)
-
     module.init(bot: name, token: token)
+    if opts[:setup_commands], do: setup_commands(module.commands(), token)
 
-    dispatcher_opts = %ExGram.Dispatcher{
-      name: name,
-      bot_info: bot_info,
-      dispatcher_name: name,
-      commands: commands,
-      regex: module.regexes(),
-      middlewares: module.middlewares(),
-      handler: {module, :handle},
-      error_handler: {module, :handle_error}
-    }
+    bot_info = get_bot_info(opts[:username], token)
+    dispatcher_opts = Dispatcher.init_state(name, bot_info, module)
 
     children = [
-      {ExGram.Dispatcher, dispatcher_opts},
+      {Dispatcher, dispatcher_opts},
       {updates_worker, {:bot, name, :token, token}}
     ]
 
     Supervisor.init(children, strategy: :one_for_one)
   end
 
-  defp maybe_fetch_bot(username, _token) when is_binary(username),
+  defp get_bot_info(username, _token) when is_binary(username),
     do: %ExGram.Model.User{username: username, is_bot: true}
 
-  defp maybe_fetch_bot(_username, token) do
+  defp get_bot_info(_username, token) do
     case ExGram.get_me(token: token) do
       {:ok, bot} -> bot
       _ -> nil
```

Comment:
It feels like `get_me` shouldn't fail (except network or token issue) and should return `ExGram.Model.User` all the times, if so we can make `bot_info` required for `Dispatcher`.